src/pybind/mgr/cephadm/inventory.py

```diff
@@ -925,6 +925,24 @@ def get_schedulable_hosts(self) -> List[HostSpec]:
             )
         ]
 
+    def get_conf_keyring_available_hosts(self) -> List[HostSpec]:
+        """
+        Returns all hosts without _no_conf_keyring label that
+        have had a refresh
+
+        Any host without that label is considered fair game for
+        a client keyring spec to match. We want to still wait
+        for refresh here so that we know what keyrings we've
+        already deployed here
+        """
+        return [
+            h for h in self.mgr.inventory.all_specs()
+            if (
+                self.host_had_daemon_refresh(h.hostname)
+                and '_no_conf_keyring' not in h.labels
```

Comment:
probably we need to declare a constant for `_no_conf_keyring` as it's used in several places

Reply:
didn't want to handle this one differently than the other special host labels, so I added a class in utils that defines them in a separate commit

Reply:
sounds reasonable :+1: